kpm/lib/kpm/base_artifact.rb

```diff
@@ -164,8 +164,10 @@ def skip_if_exists(artifact_info, coordinates, sha1_file)
         end
 
         # Finally check if remote_sha1 matches what we have locally
-        local_sha1 = Digest::SHA1.file(artifact_info[:file_path]).hexdigest
-        local_sha1 == artifact_info[:sha1]
+        if !artifact_info[:is_tgz]
+          local_sha1 = Digest::SHA1.file(artifact_info[:file_path]).hexdigest
+          local_sha1 == artifact_info[:sha1]
+        end
```

Comment:
Ruby returns nil, which is usually treated as false. Maybe it would be good to make it explicit?